Access review v2 api (#9935)
microsoftgraph/microsoft-graph-docs: api-reference/beta/resources/accessreviewinstance.md

---
title: "accessReviewInstance resource type"
description: "Represents a recurrence of an `accessReviewScheduleDefinition`."
author: "isabelleatmsft"
localization_priority: Normal
ms.prod: "microsoft-identity-platform"
doc_type: resourcePageType
---

# accessReviewInstance resource type

Namespace: microsoft.graph

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

> Represents an Azure AD [access review](accessreviewsv2-root.md) recurrence. If the parent accessReviewScheduleDefinition is a recurring access review, instances represent each recurrence. A review that does not recur will have exactly one instance. Instances also represent each unique group being reviewed in the schedule definition. If a schedule definition reviews multiple groups, each group will have a unique instance for each recurrence.

Reviewer: please make schedule definition a []()

Author: You may have cut off here?

Reviewer: Github ate my markdown.  please make schedule definition a link like \[accessReviewScheduleDefinition\]\(accessreviewscheduledefinition.md\)

| decisions | [accessReviewInstanceDecisionItem](accessreviewinstancedecisionitem.md) collection | Each user reviewed in an accessReviewInstance has a decision item representing if they were approved, denied, or not yet reviewed. |
| definition |[accessReviewScheduleDefinition](accessreviewscheduledefinition.md) | There is exactly one accessReviewScheduleDefinition associated with each instance. It is the parent schedule for the instance, where instances are created for each recurrence of a review definition and each group selected to review by the definition. |

## Relationships

| Relationship | Type	|Description|
|:---------------|:--------|:----------|
| `definition`               |[accessReviewScheduleDefinition](accessreviewscheduledefinition.md)          | There is exactly one `accessReviewScheduleDefinition` associated with each instance. It is the parent schedule for the instance, where instances are created for each recurrence of a review definition and each group selected to review by the definition. |

> | `decisions`               |[accessReviewInstanceDecisionItem](accessreviewinstancedecisionitem.md) collection        | Each user reviewed in an `accessReviewInstance` has a decision item representing if they were approved, denied, or not yet reviewed. |

Reviewer: might wish to say "representing if their access was approved, denied..."  Since it's the access being removed generally rather than the user.

---
title: "accessReviewInstance resource type"

> description: "In the Azure AD access reviews feature, the `accessReviewInstance` represents a recurrence of an `accessReviewScheduleDefinition`."

Reviewer 2: ```suggestion
description: "Represents a recurrence of an accessReviewScheduleDefinition."
```

author: "isabelleatmsft"
localization_priority: Normal
ms.prod: "microsoft-identity-platform"
doc_type: resourcePageType
---

# accessReviewInstance resource type

Namespace: microsoft.graph

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

> Represents an Azure AD [access review](accessreviewsv2-root.md) recurrence. If the parent **accessReviewScheduleDefinition** is a recurring access review, instances represent each recurrence. A review that does not recur will have exactly one instance. Instances also represent each unique group being reviewed in the schedule definition. If a schedule definition reviews multiple groups, each group will have a unique instance for each recurrence.

Reviewer 2: ```suggestion
Represents an Azure AD [access review](accessreviewsv2-root.md) recurrence. If the parent accessReviewScheduleDefinition is a recurring access review, instances represent each recurrence. A review that does not recur will have exactly one instance. Instances also represent each unique group being reviewed in the schedule definition. If a schedule definition reviews multiple groups, each group will have a unique instance for each recurrence.
```

Every **accessReviewInstance** contains a list of [decisions](accessreviewinstancedecisionitem.md) that reviewers can take action on. There is one decision per identity being reviewed.

## Methods

> | Method		   | Return Type	|Description|

Reviewer 2: ```suggestion
| Method | Return Type | Description |
```

|:---------------|:--------|:----------|
|[List accessReviewInstances](../api/accessreviewinstance-list.md) | [accessReviewInstance](accessreviewinstance.md) collection | Get a list of the [accessReviewInstance](../resources/accessreviewinstance.md) objects and their properties. |

> |[Get accessReviewInstance](../api/accessreviewinstance-get.md) | [accessReviewInstance](accessreviewinstance.md) | Returns `accessReviewInstance` for an `accessReviewScheduleDefinition`. Does not include associated `accessReviewInstanceDecisionItem`s in object. |

Reviewer 2: ```suggestion
|[Get accessReviewInstance](../api/accessreviewinstance-get.md) | [accessReviewInstance](accessreviewinstance.md) | Returns accessReviewInstance for an accessReviewScheduleDefinition. Does not include associated accessReviewInstanceDecisionItem`s in the object. |
```

> |[List pendingAccessReviewInstances](../api/accessreviewinstance-listpendingapproval.md) | [accessReviewInstance](accessreviewinstance.md) collection. | Get all pending `accessReviewInstance` resources assigned to the calling user. |

Reviewer 2: ```suggestion
|[List pendingAccessReviewInstances](../api/accessreviewinstance-listpendingapproval.md) | [accessReviewInstance](accessreviewinstance.md) collection. | Get all pending accessReviewInstance resources assigned to the calling user. |
```

> |[Send accessReviewInstance reminder](../api/accessreviewinstance-sendreminder.md) | None. | Send a reminder to the reviewers of an `accessReviewInstance`. |

Reviewer 2: ```suggestion
|[Send accessReviewInstance reminder](../api/accessreviewinstance-sendreminder.md) | None. | Send a reminder to the reviewers of an accessReviewInstance. |
```

> |[Stop accessReviewInstance](../api/accessreviewinstance-stop.md) | None. | Manually stop an `accessReviewInstance`. |

Reviewer 2: ```suggestion
|[Stop accessReviewInstance](../api/accessreviewinstance-stop.md) | None. | Manually stop an accessReviewInstance. |
```

> |[Accept recommendations](../api/accessreviewinstance-acceptrecommendations.md) | None. | Allows the calling user to accept the decision recommendation for each NotReviewed `accessReviewInstanceDecisionItem` that they are the reviewer on for a specific `accessReviewInstance`. |

Reviewer 2: ```suggestion
|[Accept recommendations](../api/accessreviewinstance-acceptrecommendations.md) | None. | Allows the calling user to accept the decision recommendation for each NotReviewed accessReviewInstanceDecisionItem that they are the reviewer on for a specific accessReviewInstance. |
```

> |[Apply decisions](../api/accessreviewinstance-applydecisions.md) | None. | Manually apply decision on an `accessReviewInstance`. |

Reviewer 2: ```suggestion
|[Apply decisions](../api/accessreviewinstance-applydecisions.md) | None. | Manually apply decision on an accessReviewInstance. |
```

## Properties

> | Property                  | Type                                |  Description |

Reviewer 2: ```suggestion
| Property | Type | Description |
```

| :-------------------------| :---------------------------------- | :---------- |

> | `id`                      |`String`                                                         | Unique identifier of the instance |

Reviewer 2: ```suggestion
| id | String | Unique identifier of the instance. |
```

> | `displayName`             |`String`                                                         | Name of the parent `accessReviewScheduleDefinition` |

Reviewer 2: ```suggestion
| displayName | String | Name of the parent accessReviewScheduleDefinition. |
```

> | `startDateTime`           |`DateTimeOffset`                                                 | DateTime when review instance is scheduled to start. May be in the future. |

Reviewer 2: ```suggestion
| startDateTime | DateTimeOffset | DateTime when review instance is scheduled to start. May be in the future. |
```

> | `endDateTime`             |`DateTimeOffset`                                                 | DateTime when review instance is scheduled to end. |

Reviewer 2: ```suggestion
| endDateTime | DateTimeOffset | DateTime when review instance is scheduled to end. |
```

> | `status`                  |`string`                                                          | Read-only. Field specifies the status of an accessReview. The typical states include `Initializing`, `NotStarted`, `Starting`, `InProgress`, `Completing`, `Completed`, `AutoReviewing`, and `AutoReviewed`. |

Reviewer 2: ```suggestion
| status | string | Specifies the status of an accessReview. The typical states include `Initializing`, `NotStarted`, `Starting`, `InProgress`, `Completing`, `Completed`, `AutoReviewing`, and `AutoReviewed`.  Read-only.|
```

> | `scope`                   |`microsoft.graph.accessReviewScope`                              | Read-only. Created based on `scope` and `instanceEnumerationScope` at the `accessReviewScheduleDefinition` level. Defines scope of users reviewed in a group. In the case of a single-group review, the scope defined at the `accessReviewScheduleDefinition` level applies to all instances. In the case of all groups review, scope may be different for each group. | 

Reviewer 2: ```suggestion
| scope | [accessReviewScope](accessreviewscope.md) | Created based on **scope** and **instanceEnumerationScope** at the accessReviewScheduleDefinition level. Defines the scope of users reviewed in a group. In the case of a single-group review, the scope defined at the `accessReviewScheduleDefinition` level applies to all instances. In the case of all groups review, scope may be different for each group. Read-only.  | 
```

> | `decisions`               |`Collection(microsoft.graph.accessReviewInstanceDecisionItem)`  | Set of decisions for this review instance. |

Reviewer 2: ```suggestion
| decisions | [accessReviewInstanceDecisionItem](accessreviewinstancedecisionitem.md) collection  | Set of decisions for this review instance. |
```

> | `definition`              |`microsoft.graph.accessReviewScheduleDefinition`                | Back link to the `accessReviewScheduleDefinition` associated with the instance. |

Reviewer 2: ```suggestion
| definition | [accessReviewScheduleDefinition](accessreviewscheduledecision.md) | Back link to the accessReviewScheduleDefinition associated with the instance. |
```

## Relationships

| Relationship | Type	|Description|
|:---------------|:--------|:----------|

> | `definition`               |[accessReviewScheduleDefinition](accessreviewscheduledefinition.md)          | There is exactly one `accessReviewScheduleDefinition` associated with each instance. It is the parent schedule for the instance, where instances are created for each recurrence of a review definition and each group selected to review by the definition. |

Reviewer 2: ```suggestion
| definition |[accessReviewScheduleDefinition](accessreviewscheduledefinition.md) | There is exactly one accessReviewScheduleDefinition associated with each instance. It is the parent schedule for the instance, where instances are created for each recurrence of a review definition and each group selected to review by the definition. |
```

> | `decisions`               |[accessReviewInstanceDecisionItem](accessreviewinstancedecisionitem.md) collection        | Each user reviewed in an `accessReviewInstance` has a decision item representing if they were approved, denied, or not yet reviewed. |

Reviewer 2: ```suggestion
| decisions | [accessReviewInstanceDecisionItem](accessreviewinstancedecisionitem.md) collection | Each user reviewed in an accessReviewInstance has a decision item representing if they were approved, denied, or not yet reviewed. |
```

## JSON representation

Here is a JSON representation of the resource.

<!-- {
  "blockType": "resource",
  "keyProperty": "id",
  "@odata.type": "microsoft.graph.accessReviewInstance",
  "baseType": "",
  "openType": false
}
-->

```json
{
 "@odata.type": "#microsoft.graph.accessReviewInstance",
 "id": "string (identifier)",
 "displayName": "string",
 "startDateTime": "string (timestamp)",
 "endDateTime": "string (timestamp)",
 "status": "string",
 "scope": {
    "@odata.type": "microsoft.graph.accessReviewScope"
  }
}
```

> ## user

Reviewer 2: This shouldn't be here. I'm not seeing where it is referenced, but it seems like a complex type that needs to be documented. It won't be able to use "user" as a name because that resource already exists.

Author: Can I name it anything I want? It is called "user" in our internal docs.
